src/nvim/ex_cmds2.c

```diff
@@ -1791,18 +1791,17 @@ void ex_args(exarg_T *eap)
   } else if (eap->cmdidx == CMD_args) {
     // ":args": list arguments.
     if (ARGCOUNT > 0) {
-      // Overwrite the command, for a short list there is no scrolling
-      // required and no wait_return().
-      gotocmdline(true);
-      for (int i = 0; i < ARGCOUNT; i++) {
-        if (i == curwin->w_arg_idx) {
-          msg_putchar('[');
+      char **items = (char **)xmalloc(sizeof(char *) * ARGCOUNT);
```

Comment:
```
../src/nvim/ex_cmds2.c: In function ‘ex_args’:
../src/nvim/ex_cmds2.c:1794:54: error: conversion to ‘long unsigned int’ from ‘int’ may change the sign of the result [-Werror=sign-conversion]
       char **items = (char **)xmalloc(sizeof(char *) * ARGCOUNT);
                                                      ^
../src/nvim/ex_cmds2.c:1803:9: error: implicit declaration of function ‘list_in_columns’ [-Werror=implicit-function-declaration]
         list_in_columns(items, ARGCOUNT, curwin->w_arg_idx);
         ^
```

What to do here?

Reply:
I think we usually do something like:

```
assert(ARGCOUNT >= 0);
char **items = xmalloc(sizeof(char *) * (size_t)ARGCOUNT);
```

Note that we don't cast `xmalloc()` result.

Reply:
Adjusted.